torchbenchmark/models/dcgan/__init__.py

```diff
@@ -0,0 +1,334 @@
+    if classname.find('Conv') != -1:
+        nn.init.normal_(m.weight.data, 0.0, 0.02)
+    elif classname.find('BatchNorm') != -1:
+        nn.init.normal_(m.weight.data, 1.0, 0.02)
+        nn.init.constant_(m.bias.data, 0)
+
+class Generator(nn.Module):
+    def __init__(self, dcgan):
+        super(Generator, self).__init__()
+        self.ngpu = dcgan.ngpu
+        self.main = nn.Sequential(
+            # input is Z, going into a convolution
+            nn.ConvTranspose2d( dcgan.nz, dcgan.ngf * 8, 4, 1, 0, bias=False),
+            nn.BatchNorm2d(dcgan.ngf * 8),
+            nn.ReLU(True),
+            # state size. (dcgan.ngf*8) x 4 x 4
+            nn.ConvTranspose2d(dcgan.ngf * 8, dcgan.ngf * 4, 4, 2, 1, bias=False),
+            nn.BatchNorm2d(dcgan.ngf * 4),
+            nn.ReLU(True),
+            # state size. (dcgan.ngf*4) x 8 x 8
+            nn.ConvTranspose2d( dcgan.ngf * 4, dcgan.ngf * 2, 4, 2, 1, bias=False),
+            nn.BatchNorm2d(dcgan.ngf * 2),
+            nn.ReLU(True),
+            # state size. (dcgan.ngf*2) x 16 x 16
+            nn.ConvTranspose2d( dcgan.ngf * 2, dcgan.ngf, 4, 2, 1, bias=False),
+            nn.BatchNorm2d(dcgan.ngf),
+            nn.ReLU(True),
+            # state size. (dcgan.ngf) x 32 x 32
+            nn.ConvTranspose2d( dcgan.ngf, dcgan.nc, 4, 2, 1, bias=False),
+            nn.Tanh()
+            # state size. (dcgan.nc) x 64 x 64
+        )
+
+        self.jt = None
+        self.jitshape = None
+        self.debug_print = False
+
+    def forward(self, input):
+
+      if self.debug_print:
+        print(input.shape)
+
+      return self.main(input)
+
+class Discriminator(nn.Module):
+    def __init__(self, ncgan):
+        ngpu = ncgan.ngpu
+        nc = ncgan.nc
+        ndf = ncgan.ndf
+
+        super(Discriminator, self).__init__()
+        self.ngpu = ngpu
+        self.main = nn.Sequential(
+            # input is (nc) x 64 x 64
+            nn.Conv2d(nc, ndf, 4, 2, 1, bias=False),
+            nn.LeakyReLU(0.2, inplace=True),
+            # state size. (ndf) x 32 x 32
+            nn.Conv2d(ndf, ndf * 2, 4, 2, 1, bias=False),
+            nn.BatchNorm2d(ndf * 2),
+            nn.LeakyReLU(0.2, inplace=True),
+            # state size. (ndf*2) x 16 x 16
+            nn.Conv2d(ndf * 2, ndf * 4, 4, 2, 1, bias=False),
+            nn.BatchNorm2d(ndf * 4),
+            nn.LeakyReLU(0.2, inplace=True),
+            # state size. (ndf*4) x 8 x 8
+            nn.Conv2d(ndf * 4, ndf * 8, 4, 2, 1, bias=False),
+            nn.BatchNorm2d(ndf * 8),
+            nn.LeakyReLU(0.2, inplace=True),
+            # state size. (ndf*8) x 4 x 4
+            nn.Conv2d(ndf * 8, 1, 4, 1, 0, bias=False),
+            nn.Sigmoid()
+        )
+        self.jt = None
+        self.jitshape = None
+
+    def forward(self, input):
+      return self.main(input)
+
+class Model(BenchmarkModel):
+    task = COMPUTER_VISION.GENERATION
+
+    def __init__(self, device=None, jit=False):
```

Comment:
In https://github.com/pytorch/benchmark/pull/535/files#diff-d6af0459a37d985953d7040c14f53feb3b9cc9e58b543aa3c2b80256d276c5e0R141 we introduced the `eval_bs` and `train_bs` to support adjusting batch size via `run.py`, can you add parameters to `__init__()` so that we can also test different batches sizes for dcgan through `run.py`?